src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -337,36 +363,37 @@ def _construct_local_view(self, field: MemletExpr | DataExpr) -> DataExpr:
                 view_shape = tuple(desc.shape[i] for i in local_dim_indices)
                 view_strides = tuple(desc.strides[i] for i in local_dim_indices)
             view, _ = self.sdfg.add_view(
-                f"{field.node.data}_view",
+                f"{field.dc_node.data}_view",
                 view_shape,
                 desc.dtype,
                 strides=view_strides,
                 find_new_name=True,
             )
             local_view_node = self.state.add_access(view)
-            self._add_input_data_edge(field.node, field.subset, local_view_node)
+            self._add_input_data_edge(field.dc_node, field.subset, local_view_node)
 
-            return DataExpr(local_view_node, desc.dtype)
+            return ValueExpr(local_view_node, desc.dtype)
 
         else:
             return field
 
     def _construct_tasklet_result(
         self,
-        dtype: dace.typeclass,
+        dc_dtype: dace.typeclass,
         src_node: dace.nodes.Tasklet,
         src_connector: str,
+        local_offset: Optional[str] = None,
```

Comment:
You have these classes that combines certain things like `FieldOpsData`, why are you passing the data individually here and not encapsulated inside them?

Or is this function too low level?

Reply:
I cannot use `FieldopData` in this case. `FieldopData` is supposed to contain fields with global domain. The tasklet result constructed here is in the local domain: by that I mean it returns just a scalar that represents an intermediate result of an element-wise operation.